### PR TITLE
dbt Cloud: add on-premises setup and configuration guide

### PR DESCRIPTION
**Pull request by cmcarthur**
## Description & motivation
We have a Google doc that contains the on-premises setup and installation guide. There's no reason that should live in a Google doc, and in some cases it actually causes confusion. Docs is a better place for this to live.

### CODE REVIEW

**Comment by clrcrl** [comment]
@drewbanin — at some point we should consider whether we want to pull dbt Cloud docs into a different header

**Comment by cmcarthur** [comment]
New section preview is here: https://deploy-preview-85--docs-getdbt-com.netlify.com/docs/dbt-cloud/on-premises/index

**Comment by drewbanin** [comment]
@clrcrl I would love to move dbt Cloud docs into a different header! Are we going to have too many headers/

**Comment by drewbanin** [comment]
?

File: website/docs/docs/dbt-cloud/on-premises/setup.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+A new version of dbt Cloud will appear on the Version History page in your Configuration Console anytime any of the following happen:
+
+- A new version of the dbt Cloud code is released. This typically happens once per week, and each new version will be accompanied by a changelog.
+- Any configuration change is applied to your application via the Configuration Console.
+- Anytime an edit is applied to your Kubernetes configs via the overlays mechanism built into kots.
+
+You can apply or roll back these changes at any time by clicking the Deploy and Rollback buttons on the right side of this screen.
+
+### Github
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/on-premises/setup.md:70]: You might want to note that this is only applicable to GitHub's self-hosted offering!

Comment (cmcarthur) [reply website/docs/docs/dbt-cloud/on-premises/setup.md:70]: @drewbanin that's not true! for onprem, customers can create their own Github app even in Github cloud-hosted

**Pull request merged**